mmdet/core/hook/sync_bn_hook.py

```diff
@@ -0,0 +1,126 @@
+from mmcv.runner.hooks import HOOKS, Hook
+from mmcv.runner import get_dist_info
+from collections import OrderedDict
+
+import torch
+from torch import distributed as dist
+from torch import nn
+import functools
+import pickle
+
+
+ASYNC_NORM = (
+    nn.BatchNorm1d,
+    nn.BatchNorm2d,
+    nn.BatchNorm3d,
+    nn.InstanceNorm1d,
+    nn.InstanceNorm2d,
+    nn.InstanceNorm3d,
+)
+
+
+def get_async_norm_states(module):
+    async_norm_states = OrderedDict()
+    for name, child in module.named_modules():
+        if isinstance(child, ASYNC_NORM):
+            for k, v in child.state_dict().items():
+                async_norm_states[".".join([name, k])] = v
+    return async_norm_states
+
+
+def pyobj2tensor(pyobj, device="cuda"):
```

Comment:
I recommend moving these functions to `mmdet/utils`, it is general enough to be a basic function